### ISSUE TITLE
VPN - Allows termination over IP without specifying device

### ISSUE THREAD

**Issue by mlazzarotto** [netbox-community/netbox#14727]
### NetBox version

v3.7.0

### Feature type

Change to existing functionality

### Proposed functionality

I open this issue to ask about allowing VPN termination on an IP address without having to specify a device and its interface.

This is a simple mockup of the UI change.
![image](https://github.com/netbox-community/netbox/assets/8932945/f558357a-be56-40dd-ac1f-d08c80f8b734)

### Use case

The typical use case is where the other side of the VPN is terminated on a customer/vendor/contractor's device but the device and interface details are unknown.
Also, in this way, we would not be forced to create dummy devices that would end up staying on Netbox even though they are not actually owned by us.

**Comment by markkuleinio** [comment]
This is actually our use case as well: no need to save the remote-side device details, just the termination IP address because that's the remote-side detail we need when provisioning the tunnel on our side.

**Comment by jeremystretch** [comment]
This would have been a topic for discussion during the [v3.7 beta valuation period](https://github.com/netbox-community/netbox/discussions/14430) last month. Unfortunately, now that v3.7 has been released we are no longer considering major changes to the VPN tunnels implementation.

However, please note that it is perfectly valid to create only one termination on a tunnel; you can use a custom field to record the IP address of the remote end for which complete termination details are unknown.

**Comment by markkuleinio** [comment]
I've been testing with custom fields with VPN tunnels but the challenge with them is that they create (as far as I see it) a one-way relationship, meaning that if I configure a custom field in the VPN tunnel with the IP address object, I cannot trace back where the IP address is used. Also search does not find IP addresses in custom fields. (The same goes with contacts that I thought using in tunnels.)

Using tags could be another option.

**Comment by alehaa** [comment]
How about meeting halfway? Changing the termination to nullable might break something or have other consequences. But there's also the concept of circuits for external connections. Why not just add a circuit as a possible endpoint for tunnels?

**Comment by jeremystretch** [comment]
The proposal here (I believe) is to allow creating a tunnel termination with an outside IP address defined, but no termination interface.

**Comment by antoinekh** [comment]
We had the same needs and we decide to create a fake device called “dummy_vpn_peer” that host all the interfaces for the tunnel termination of the remote peer

**Comment by mlazzarotto** [comment]
> The proposal here (I believe) is to allow creating a tunnel termination with an outside IP address defined, but no termination interface.

Yes, that's correct. We have more than one hundred site-to-site VPNs, most of them are with external vendors or customers which we don't manage the firewalls of. Therefore I think that cluttering our Netbox with hundreds of "dummy_customer/vendor_peer" devices would be counterproductive and would increase the management difficulty of Netbox, and also slow down the searches.

**Comment by sleepinggenius2** [comment]
I just came across this situation as well and was thinking about potential solutions. Since the `termination` field on the `TunnelTermination` model is already a GFK, I like the idea of just extending it to support more models. A suggestion was already make for `Circuit`, but I'm thinking `ProviderNetwork` and maybe `Tenant` would also be good candidates. In our environment, that additional context would be a lot more useful than just a null value. For example, being able to go to a `ProviderNetwork` instance and see all the tunnels we have going to them would be super helpful.
